src/CoreFoundation/Dispatch.cs

```diff
@@ -177,6 +200,17 @@ public DispatchQueue (string label, bool concurrent)
 				throw new Exception ("Error creating dispatch queue");
 		}
 		
+		[Mac (10,12)]
+		[iOS (10,0)]
+		[TV (10,0)]
+		[Watch (3,0)]
+		public DispatchQueue (string label, Attributes attributes, DispatchQueue target = null)
+			: base (dispatch_queue_create_with_target (label, attributes?.Create () ?? IntPtr.Zero, target.GetHandle ()), true)
+		{
+			if (Handle == IntPtr.Zero)
+				throw new Exception ("Error creating dispatch queue");
```

Comment:
with your last PR (using `NativeObject`) did not we _inherit_ a null handle check with `InitializeHandle` ?